CHANGELOG.md

```diff
@@ -159,7 +159,9 @@ Previously pre-approved signatures relying on the `msg.sender` variable couldn't
 
 #### `encodeTransactionData` public method removal
 
-To fit all of the above changes into the bytecode size limit, the `encodeTransactionData` method was made private.
+PR: [#603](https://github.com/safe-global/safe-smart-account/pull/603)
+
+To fit all of the above changes into the bytecode size limit, the `encodeTransactionData` method was made private. Please note that in the subsequent PR [#847](https://github.com/safe-global/safe-smart-account/pull/847), the function `encodeTransactionData` has been completely removed and replaced with an optimized and inlined assembly version.
```

Comment:
```suggestion
To fit all of the above changes into the bytecode size limit, the `encodeTransactionData` method was removed, and in the PR [#847](https://github.com/safe-global/safe-smart-account/pull/847), the function `getTransactionHash` which had made use of the function, was refactored with an optimized inline-assembly version.
```

hey @pcaversaccio, thank you very much for the pr! since 1.5.0 hasn't been released yet, the changelog isn't final, so it's fine if we drop the public->private change and just state that it was removed in favour of the optimized version.

Reply:
> To fit all of the above changes into the bytecode size limit, the `encodeTransactionData` method was removed

I think this is wrong, no? It was made `private` due to the bytecode size limit. Later it was removed for in an assembly version due to optimisation reasons. Thus, I think the initial wording is still better.

Or alternatively, how about this?

```md
The `encodeTransactionData` function has been refactored in two stages since the last release. Initially, due to bytecode size constraints, it was modified to a `private` function. Subsequently, in PR [#847](https://github.com/safe-global/safe-smart-account/pull/847), `encodeTransactionData` was entirely removed and replaced with an optimized, inline-assembly implementation within the `getTransactionHash` function.
```

Reply:
The alternative version is perfect, thank you. Could you please update the PR with it?

Reply:
Pushed via https://github.com/safe-global/safe-smart-account/pull/855/commits/b1088664faf36dfd36707f9ec9a1b99131511418. I amended the proposed version slightly to add the link the to PR:

```diff
- it was modified to a private function.
+ it was modified in PR [#603](https://github.com/safe-global/safe-smart-account/pull/603) to a `private` function.
```